.github/workflows/post-merge-push-test-container-to-build.yml

```diff
@@ -0,0 +1,53 @@
+name: Test Container Build and Deploy - Build
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch
+  push:
+    branches:
+      - main
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+env: # Only adding the variables in that are required for 
+  AWS_REGION: eu-west-2
+
+jobs:
+  deploy-test-container-to-build-ecr:
+    name: Validate & Deploy Test Container to Build Envrionment ECR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v4.0.2
+        with:
+          role-to-assume: ${{ secrets.DEV_CRI_IPR_GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      # ... Build and validate SAM application ...
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@main
+        with:
+          cosign-release: 'v1.9.0'
+
+      - name: Build, tag, and push testing image to Amazon ECR
+        env:
+          BUILD_CONTAINER_SIGN_KMS_KEY: ${{ secrets.BUILD_CONTAINER_SIGN_KMS_KEY }}
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          BUILD_TEST_ECR_REPOSITORY: ${{ secrets.BUILD_FRONT_TEST_ECR_REPOSITORY }}
+          IMAGE_TAG: latest
+        run: |
+          docker build -f Dockerfile.test -t $ECR_REGISTRY/$BUILD_TEST_ECR_REPOSITORY:$IMAGE_TAG .
```

Comment:
So if the tests that are run from the Dockerfile fail, does that GHA fail or the code pipeline? Just trying to understand how this will stop promotion of broken code to higher envs

Reply:
How does it relate to the pipeline? This workflow builds a test container and pushes it to the ECR.